subprojects/core/src/main/java/org/gradle/internal/service/scopes/ExecutionGlobalServices.java

```diff
@@ -156,20 +148,17 @@ TypeAnnotationMetadataStore createAnnotationMetadataStore(CrossBuildInMemoryCach
                 "kotlin"
             ),
             ImmutableSet.of(
-                ClosureBackedAction.class,
+                // Getter called when there is a nested bean with action in a Task
                 ConfigureUtil.WrappedConfigureAction.class,
-                Describable.class,
+                // Inputs can extend these classes,
+                // so we want to ignore get methods declared in these classes
```

Comment:
This doesn't sound right as an explanation. These are internal types, and we do extend them in Gradle code, but third party code should not do the same. Instead it should call public methods on `ObjectFactory` etc.

Shouldn't we mark getters on the implementations instead as `@Internal`? These _are_ domain objects, so they should be aware of the build model, and thus inputs and outputs.

Reply:
We can annotate them with `@Internal` yes.